### PR TITLE
Closes MAIN-34

### PR DESCRIPTION
**Pull request by charless-ifit**
This pull request closes [MAIN-34](https://ifitdev.atlassian.net/browse/MAIN-34)

If for some reason there is no internet connectivity when trying to log analytics then we will cache them and try to post them again once we're connected.

### CODE REVIEW

**Review by chrisr-ifit — CHANGES_REQUESTED** [review]
Not working as expected in wolf for iOS, see https://github.com/ifit/wolf/pull/4005

**Review by iFitMicahN — CHANGES_REQUESTED** [review]
I get 91 builds errors on your branch for this project. I didn't get them on `master`. It sounds like what Jace ran into but he posted that on the Palantir PR not this one...

Any idea what the deal is? I'll dig in soon but thought I'd go ahead and ask.

File: Analytics.Xamarin.Pcl/Client.cs [hunk 1]
```diff
@@ -232,9 +232,9 @@ public Task Group(string userId, string groupId, IDictionary<string, object> tra
 		/// that it is in human readable form. For example, "Bought T-Shirt"
 		/// or "Started an exercise"</param>
 		///
-		public Task Track(string userId, string eventName)
+		public async Task<bool> Track(string userId, string eventName)
```
Comment (iFitMicahN) [inline Analytics.Xamarin.Pcl/Client.cs]: I don't think you needed to change these. Just doing `Task`->`Task<bool>` without adding the `async` or `await` is enough, which is what you did for most of the others it looks like. I'd prefer that it matched the other so the file was consistent throughout but it isn't a huge deal.

**Review by iFitMicahN — APPROVED** [review]
conceding the formatting :)

**Review by bendowden — APPROVED** [review]

**Comment by charless-ifit** [comment]
Merged

**Pull request merged**